clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3192,6 +3192,9 @@ def warn_dllimport_dropped_from_inline_function : Warning<
   InGroup<IgnoredAttributes>;
 def warn_attribute_ignored : Warning<"%0 attribute ignored">,
   InGroup<IgnoredAttributes>;
+def warn_attribute_on_direct_kernel_callee_only : Warning<"%0 attribute allowed"
+  " only on function directly called from kernel; attribute ignored">,
```

Comment:
only on function -> only on a function

Reply:
"called from kernel" is also a bit awkward. called from a kernel? called from a kernel function?

Reply:
OK